misc/hamming_distance.c

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file
+ * @brief [Hamming distance](https://en.wikipedia.org/wiki/Hamming_distance)
+ * algorithm implementation.
+ * @details
+ * In information theory, the Hamming distance between two strings of
+ * equal length is the number of positions at which the corresponding symbols
+ * are different.
+ * @author [Aybars Nazlica](https://github.com/aybarsnazlica)
+ */
+
+#include <assert.h>
+#include <stdio.h>
+
+/**
+ * @brief Function to calculate the Hamming distance between two strings
+ * @param param1 string 1
+ * @param param2 string 2
+ * @returns Hamming distance
+ */
+int hammingDistance(char* str1, char* str2)
+{
+    int i = 0, distance = 0;
+
+    while (str1[i] != '\0')
+    {
+        if (str1[i] == str2[i])
```

Comment:
Is it just me or shouldn't this be str1[i] != str2[i]? Both asserts failed when I tried to run this code
```suggestion
        if (str1[i] != str2[i])
```

Reply:
Right, thanks.